OpenSHMEM 1.6rc1 Draft (#540)
openshmem-org/specification: content/shmem_collect.tex

    \openshmem \FUNC{collect} and \FUNC{fcollect} routines perform a collective
    operation to concatenate \VAR{nelems}
    data items from the \source{} array into the

>     \dest{} array, over an \openshmem team in processor number order. 

Author: ```suggestion
    \dest{} array, over an \openshmem team in process number order. 
```

    If \VAR{team} compares equal to \LibConstRef{SHMEM\_TEAM\_INVALID} or is
    otherwise invalid, the behavior is undefined.

    Before any \ac{PE} calls a collect routine, the following conditions must
    be ensured, otherwise the behavior is undefined:
    \begin{itemize}
        \item The \dest{} array on all \acp{PE} in the team is ready to
            accept the result of the operation.
        \item The \source{} array at the local \ac{PE} is ready to be read
            by any \ac{PE} in the team.
    \end{itemize}
    The application does not need to synchronize to ensure that the \source{}
    array is ready across all \acp{PE} prior to calling this routine.

\begin{DeprecateBlock}
    \openshmem \FUNC{collect} and \FUNC{fcollect} routines perform a collective
    operation to concatenate \VAR{nelems}
    data items from the \source{} array into the
    \dest{} array, over an \openshmem active set

>     in processor number order. The resultant \dest{} array contains the contribution from

Author: ```suggestion
    in process number order. The resultant \dest{} array contains the contribution from
```

Author: @jdinan Search for any other instances of "processor".

    only \acp{PE} in the active set call the routine. If a \ac{PE} not in the
    active set and calls this collective routine, the behavior is undefined.

    The values of arguments \VAR{PE\_start}, \VAR{logPE\_stride}, and \VAR{PE\_size}
    must be the same value on all \acp{PE} in the active set. The same
    \VAR{pSync} work array must be passed by all \acp{PE} in the active set.

    Upon return from a collective routine, the following are true for the local
    \ac{PE}:
    \begin{itemize}
    \item The \dest{} array is updated and the \source{} array may be safely reused.
    \item For active-set-based collective routines, the values in the \VAR{pSync} array are
    restored to the original values.
    \end{itemize}
\end{DeprecateBlock}
}

\apireturnvalues{
    Zero on successful local completion. Nonzero otherwise.
}

\apinotes{
\begin{DeprecateBlock}
    The collective routines operate on active \ac{PE} sets that have a
    non-power-of-two \VAR{PE\_size} with some performance degradation.  They operate
    with no performance degradation when \VAR{nelems} is a non-power-of-two value.
\end{DeprecateBlock}
    The collective routines that operate on teams containing a

>     non-power-of-two of PEs do so with some performance degradation. They operate

Author: ```suggestion
    non-power-of-two of \acp{PE} do so with some performance degradation. They operate
```
@jdinan Run a search and replace on instances of "PE" and "PEs" that need formatting.
